src/utils/tensor.js

```diff
@@ -28,27 +30,40 @@ const DataTypeMap = new Map([
     ['int32', Int32Array],
     ['uint32', Uint32Array],
     ['int64', BigInt64Array],
-])
+]))
 
 /**
  * @typedef {import('./maths.js').AnyTypedArray | any[]} DataArray
  */
 
 const ONNXTensor = ONNX.Tensor;
 
-export class Tensor extends ONNXTensor {
+export class Tensor {
+    /** @type {DataType} Type of the tensor. */
+    type;
+
+    /** @type {DataArray} The data stored in the tensor. */
+    data;
```

Comment:
ONNX has this order:

```js
Object.keys(new onnx.ONNX.Tensor('string', ['a', 'b', 'c', 'd']))
```

![image](https://github.com/xenova/transformers.js/assets/5236548/aeff2828-80cb-4f6a-879f-3c92e23728e7)

In short, `dims` goes to the top while adding `size` (a number) at the end.

(this ensures internal object alignment, so `JSON.stringify` stays compatible)